target/linux/layerscape/image/mkits-multiple-config.sh

```diff
@@ -0,0 +1,594 @@
+			description = "Flattened Device Tree blob${name}";
+			data = /incbin/("${image}");
+			type = "flat_dt";
+			arch = "${ARCH}";
+			load = <${loadaddr}>;
+			compression = "none";
+EOF
+	emit_cksum ${checksum}
+	if [ -z "$SIGN_IN_CONF" ] ; then
+		emit_signature "" "$7" "" "" "$6" ""
+	fi
+	echo "		};" >> ${OUTPUT}
+}
+
+#
+# Emit ramdisk node
+#
+emit_ramdisk() {
+	local image=${1}
+	local count=${2:-${MAX_IMG}}
+	local compresson=${3:-none}
+	local checksum=${4:-sha1}
+	local name=${5}
+
+	[ -z "${name}" ] || name=" ${name}"
+	cat << EOF >> ${OUTPUT}
+		ramdisk@${count} {
+			description = "ramdisk${name}";
+			data = /incbin/("${image}");
+			type = "ramdisk";
+			arch = "${ARCH}";
+			os = "linux";
+			compression = "${compresson}";
+EOF
+	emit_cksum ${checksum}
+	if [ -z "$SIGN_IN_CONF" ] ; then
+		emit_signature "" "" "$7" "" "" "$6"
+	fi
+	echo "		};" >> ${OUTPUT}
+}
+
+#
+# Emit check sum sub node
+#
+emit_cksum() {
+	csum_list=$@
+	count=1
+	for csum in ${csum_list}; do
+		cat << EOF >> ${OUTPUT}
+			hash@${count} {
+				algo = "${csum}";
+			};
+EOF
+		count=`expr ${count} + 1`
+	done
+}
+
+#
+# Emit signature sub node
+#
+emit_signature() {
+	local kernel=$1
+	local fdt=$2
+	local rootfs=$3
+	local kernel_key=$4
+	local fdt_key=$5
+	local rootfs_key=$6
+	local imgs=""
+	local count=0
+	local chk_list="" algo="" algos="" i=""
+
+	for i in kernel fdt rootfs; do
+		eval algo=\$$i
+		eval key=\$${i}_key
+		[ -n "$algo" ] || continue
+		if ! echo "$algos" | grep -q $algo; then
+			if [ -z "$algos" ]; then
+				algos=$algo
+			else
+				algos="${algos} $algo"
+			fi
+		fi
+		if ! echo "$keys" | grep -q $key; then
+			if [ -z "$keys" ]; then
+				keys=$key
+			else
+				keys="${keys} $key"
+			fi
+		fi
+	done
+
+	for algo in $algos; do
+		for key in $keys; do
+			img=""
+			for i in kernel fdt rootfs; do
+				eval tmp_algo=\$$i
+				eval tmp_key=\$${i}_key
+				[ "$tmp_algo" == "$algo" ] || continue
+				[ "$tmp_key" == "$key" ] || continue
+				if [ -z "$img" ]; then
+					img=$i
+				else
+					img=${img},$i
+				fi
+			done
+
+			[ -n "$img" ] || continue
+			cat << EOF >> ${OUTPUT}
+			signature@${count} {
+				algo = "${algo}";
+				key-name-hint = "${key}";
+EOF
+			if [ -n "$SIGN_IN_CONF" ] ; then
+				echo "			sign-images = \"$img\";" >> ${OUTPUT}
+			fi
+			echo "			};" >> ${OUTPUT}
+
+			count=`expr ${count} + 1`
+		done
+	done
+}
+
+#
+# Emit config sub nodes
+#
+emit_config() {
+	local conf_csum="sha1"
+
+	config_name="conf@${1}"
+	if [ ! -z "${11}" ]; then
+		config_name="${11}"
+	fi 
+	if [ -z "${2}" ]; then
+		echo "Error: config has no kernel img, skipping conf node!"
+		return 0
+	fi
+
+	# Test if we have any DTBs at all
+	if [ -z "${3}" ] ; then
+		conf_desc="Boot Linux kernel"
+		fdt_line=""
+	else
+		conf_desc="Boot Linux kernel with FDT blob"
+		fdt_line="
+			fdt = \"fdt@${3}\";"
+	fi
+
+	# Test if we have any ROOTFS at all
+	if [ -n "${4}" ] ; then
+		conf_desc="$conf_desc + ramdisk"
+		fdt_line="${fdt_line}
```

Comment:
Ditto.  Plus you don't need curly braces.

Curly braces are mostly useful for a complex interpolation like `${name:+...}` so you know where the end of the interpolation is, or else when concatenating strings to variable values like `${var}suffix` where you don't want to look up `$varsuffix` as a variable name.